core/templates/services/rte-helper-modal.controller.ts

```diff
@@ -245,70 +253,86 @@ export class RteHelperModalComponent {
       }
     }
     this.ngbActiveModal.dismiss(true);
+    this.rteFieldValues.unsubscribe();
   }
 
   delete(): void {
     this.ngbActiveModal.dismiss(true);
+    this.rteFieldValues.unsubscribe();
   }
 
-  disableSaveButtonForMathRte(): boolean {
-    // This method disables the save button when the Math SVG has not yet
-    // been generated but being processed.
-    if (!this.currentRteIsMathExpressionEditor) {
-      return false;
-    } else {
-      // We know that this is a math rich text component. Hence we can make the
-      // the type more specific.
-      const { value } = this.tmpCustomizationArgs[0] as Extract<
-        CustomizationArgsNameAndValueArray[number],
-        { name: 'math_content' }
-      >;
-      return value.mathExpressionSvgIsBeingProcessed || value.raw_latex === '';
-    }
-  }
-
-  disableSaveButtonForLinkRte(): boolean {
-    // This method disables the save button when the `text` field for the
-    // Link RTE looks like a URL but it does not match the `url`. Otherwise,
-    // creators can make the `url` a malicious website and make the `text`
-    // a safe website.
-    if (!this.currentRteIsLinkEditor) {
-      return false;
-    }
-    // We know that this is a link rich text component. Hence we can make the
-    // the type more specific.
-    const tmpCustomizationArgs = this.tmpCustomizationArgs as Extract<
-      CustomizationArgsNameAndValueArray[number],
-      { name: 'url' | 'text' }
-    >[];
-    let url: string = tmpCustomizationArgs[0].value;
-    let text: string = tmpCustomizationArgs[1].value;
+  updateRteSaveButtonState(value: number | string | boolean): void {
+    if (this.componentId === this.COMPONENT_ID_MATH) {
+      let rawLatex: string = value[0].raw_latex;
+      let mathExpressionSvgIsBeingProcessed: boolean = (
+        value[0].mathExpressionSvgIsBeingProcessed);
+      this.saveButtonIsDisabled = (
+        mathExpressionSvgIsBeingProcessed || rawLatex === '');
+    } else if (this.componentId === this.COMPONENT_ID_VIDEO) {
+      let start: number = value[1];
+      let end: number = value[2];
+      if (start === 0 && end === 0) {
+        this.saveButtonIsDisabled = false;
+      } else {
+        this.saveButtonIsDisabled = start >= end;
+      }
+      this.updateRteErrorMessage();
+    } else if (this.componentId === this.COMPONENT_ID_LINK) {
+      let url: string = value[0];
+      let text: string = value[1];
+      // First check if the `text` looks like a URL.
+      const suffixes = ['.com', '.org', '.edu', '.gov'];
+      let urlSuffixIsValid = false;
 
-    // First check if the `text` looks like a URL.
-    const suffixes = ['.com', '.org', '.edu', '.gov'];
-    let textLooksLikeUrl = false;
-    for (const suffix of suffixes) {
-      if (text.endsWith(suffix)) {
-        textLooksLikeUrl = true;
+      for (const suffix of suffixes) {
+        if (url !== '' && url.endsWith(suffix)) {
+          urlSuffixIsValid = true;
```

Comment:
I think you want to keep the previous name. The idea is to "autofill" the URL if the text looks like a URL. We aren't validating the URL itself; many URLs have other suffixes.

Reply:
Can I simply delete the URL validation then? Since I already handle it to set the text to the URL if no input is given in the else clause

Reply:
Why do you say it's not used? As far as I can tell urlSuffixIsInvalid/Valid is used in an if clause just below, but I might be missing something here.

Reply:
I mean that since we do not want to validate the suffix of URLs, we could just skip this part and validate the url-field with the text-field. The if-clause here used to validate if the ending of the URL was valid, and if it was not ending in any of the suffixes the Done button would be deactivated. The previous code had similar code, but no validation of the URL was actually performed hence why I think we can simply delete this URL validation if we do not need it (and only validate the url-field vs the text-field to see if it matches or not). Please let me know if you want me to add back the URL validation (which I think then should have some kind of affect on the done-button, or if we can just remove this), thanks

Reply:
Ok seems like i misunderstood the validation in the first part and thought it was to see if the url was valid, now it makes sense again so changed it back to the previous behaviour